### ISSUE TITLE
QTML honorifics

### ISSUE THREAD

**Issue by mustafa0x** [quranacademy/digital-quran-docs#11]
- واٰله  should be removed from صَلّى اللهُ عليهِ واٰله وسلّم, as ﷺ is the more standard honorific. صَلّى اللهُ عليهِ واٰله وسلّم will be added to Unicode 14, but it is more rarely used.
- رحمة الله عليه (and its variations) should be replaced with رحمه الله. Similar to above, it is far more common, and it has been accepted by Unicode.
- The honorifics mentioned have all been accepted by Unicode. They should be included in Unicode 14 (1.5 years from now). However, they have been given code points that will almost certainly not change (I've confirmed this with a Unicode member). Thus, using the code points themselves makes more sense.

- ﷺ
- ﷻ
- U+FDFE - سبحانه وتعالى
- U+FDFF - عز وجل
- U+FD40 - رحمه الله
- U+FD41 - رضي الله عنه
- U+FD42 - رضي الله عنها
- U+FD43 - رضي الله عنهم
- U+FD44 - رضي الله عنهما
- U+FD45 - رضي الله عنهن
- U+FD47 - عليه السلام
- U+FD4D - عليها السلام
- U+FD48 - عليهم السلام
- U+FD49 - عليهما السلام
- U+FD4A - عليه الصلاة والسلام

**Comment by rguliev** [comment]
> - واٰله should be removed from صَلّى اللهُ عليهِ واٰله وسلّم, as ﷺ is the more standard honorific. صَلّى اللهُ عليهِ واٰله وسلّم will be added to Unicode 14, but it is more rarely used.
> - رحمة الله عليه (and its variations) should be replaced with رحمه الله. Similar to above, it is far more common, and it has been accepted by Unicode.

Sure. Thanks a lot! Could you please check all Arabic texts of honorifics and make a pull request?

> The honorifics mentioned in QTML have all been accepted by Unicode. They should be included in Unicode 14 (1.5 years from now). However, they have been given code points that will almost certainly not change (I've confirmed this with a Unicode member). Thus, using the code points themselves makes more sense.

We had a discussion on this and ended up with XML for the following reasons:

- Even existing Unicode symbols are not supported perfectly. For example, here is how I see your message:
 
> ![Screenshot from 2019-12-20 16-51-35](https://user-images.githubusercontent.com/9852534/71259239-09aae180-2349-11ea-8c3d-d59f4032c4d9.png)

- Thus, for the new Unicode symbols, it is not just 1.5 years from now, rather it is 1.5 years + X years for the proper support by the most common fonts.
- Adding a new honorific becomes very difficult and time-consuming: discussion with Unicode team, reserve a Unicode-symbol, etc. In contrast, XML format allows simple and flexible manipulation. 
- We want to give some flexibility in the rendering: some might prefer custom glyphs, some just print as a text in parentheses, etc. Of course, one can always search for a Unicode-symbol and replace it, but then it breaks the logic of QTML: some parts would be searched by tags, the others by regexs. Quite confusing.

**Comment by mustafa0x** [comment]
I don't feel like those arguments apply, since we're talking about the source, not the final distributed file.

> Of course, one can always search for a Unicode-symbol and replace it, but then it breaks the logic of QTML: some parts would be searched by tags, the others by regexs. Quite confusing.

In general, semantic Unicode is superior to XML. And replacing a code point doesn't need a regex, but a dumb string replacement.

**Comment by rguliev** [comment]
> I don't feel like those arguments apply since we're talking about the source, not the final distributed file.

Sorry, maybe I confused you, QTML is a format for the final distribution. I should write it explicitly in the docs.

**Comment by mustafa0x** [comment]
QTML must be converted to a final format (PDF, HTML, or system-ui) for user consumption. During that conversion you can address the issues you mentioned.

**Comment by rguliev** [comment]
Exactly! Web and mobile developers are supposed to be major users. We intend to add helper tools for their convenience. See my comment here #13
